sample/spi-sample/README.md

```diff
@@ -1,12 +1,14 @@
 # Carbon Datasource SPI Sample
 
-This is a sample java client application which uses WSO2 carbon-datasources services exposed through Java SPI to access them in non-OSGi environment. The JNDI support is not there for carbon datasources to be used in non-OSGi environment at the moment.
+This is a sample java client application which uses WSO2 carbon-datasources services exposed through Java SPI to access them in non-OSGi environment. The carbon datasources are supported with In-memory JNDI context service provider to be used in non-OSGi environment by default via carbon-jndi and if needed a custom JNDI context provider can be plugged in to use with carbon-datasources as explained below.
 
 ## Using carbon datasources in non-OSGi environment
 
 The datasources required for non-OSGi client application can be defined in a configuration file and naming convention of the configuration file is *-datasources.xml. Refer the sample configuration file as follows;
 JNDI configuration cannot be added to a datasource because JNDI support is not there for non-OSGi applications at the moment.
```

Comment:
This needs to be corrected. With this PR, we will be able to change the JNDI configuration.